airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/components/debezium/DebeziumComponent.kt

```diff
@@ -0,0 +1,389 @@
+     * message as global CDC state.
+     */
+    @JvmRecord
+    data class State(val offset: Offset, val schema: Optional<Schema>) {
+
+        /** [Offset] maps to the Debezium offset file contents. */
+        @JvmRecord data class Offset(val debeziumOffset: Map<JsonNode, JsonNode>)
+
+        /** [Schema] maps to the Debezium schema history file contents. */
+        @JvmRecord data class Schema(val debeziumSchemaHistory: List<HistoryRecord>)
+    }
+
+    /**
+     * [Output] is the return type of [collect] and contains:
+     * - [data] contains the [Record] objects,
+     * - [state] has the contents of the Debezium offset and schema history files, see [State]
+     * - [executionSummary] has statistics on the execution of [collect] like runtime, etc, see
+     * [ExecutionSummary],
+     * - [completionReasons] is the set of reasons of what cause [collect] to stop, see
+     * [CompletionReason].
+     */
+    @JvmRecord
+    data class Output(
+        val data: Sequence<Record>,
+        val state: State,
+        val executionSummary: ExecutionSummary,
+        val completionReasons: Set<CompletionReason>
+    )
+
+    /** [Record] wraps a Debezium change data event. */
+    @JvmRecord
+    data class Record(val debeziumEventValue: JsonNode) {
+
+        /** True if this is a Debezium heartbeat event. These aren't persisted in [Output]. */
+        val isHeartbeat: Boolean
+            get() = kind() == Kind.HEARTBEAT
+
+        /** The datum prior to this event; null for insertions. */
+        fun before(): JsonNode {
+            return element("before")
+        }
+
+        /** The datum following this event; null for deletions. */
+        fun after(): JsonNode {
+            return element("after")
+        }
+
+        /** Metadata containing transaction IDs, LSNs, etc, null for heartbeats. */
+        fun source(): JsonNode {
+            return element("source")
+        }
+
+        /**
+         * Convenience function for accessing child object nodes of the debezium event root node.
+         */
+        fun element(fieldName: String?): JsonNode {
+            if (!debeziumEventValue.has(fieldName)) {
+                return NullNode.getInstance()
+            }
+            return debeziumEventValue[fieldName]
+        }
+
+        /** Change data event [Kind]. */
+        fun kind(): Kind {
+            val source = source()
+            if (source.isNull) {
+                return Kind.HEARTBEAT
+            }
+            val snapshot = source["snapshot"] ?: return Kind.CHANGE
+            return when (snapshot.asText().lowercase(Locale.getDefault())) {
+                "false" -> Kind.CHANGE
+                "last" -> Kind.SNAPSHOT_COMPLETE
+                else -> Kind.SNAPSHOT_ONGOING
+            }
+        }
+
+        /** Kinds of Debezium change data events. */
+        enum class Kind {
+            /** [SNAPSHOT_ONGOING] is a change data event part of a Debezium Snapshot. */
+            SNAPSHOT_ONGOING,
```

Comment:
I don't think any of our connectors use debezium for snapshotting now - we can probably remove support for those events to simplify the API

Reply:
That's wonderful! Yes let's.